Tools/autotest/common.py

```diff
@@ -5746,6 +5746,7 @@ def wait_heartbeat(self, drain_mav=True, quiet=False, *args, **x):
         x["timeout"] = 1
         tstart = time.time()
         while True:
+            self.drain_all_pexpects()
```

Comment:
can't this make use delay far more that just a heartbeat ?

Reply:
`self.mav.wait_heartbeat` blocks waiting for a heartbeat.